types/index.d.ts

```diff
@@ -0,0 +1,166 @@
+/// <reference types="node" />
+
+declare function Postgres<T extends Postgres.CustomTypeList = {}>(options?: Postgres.Options<T>): Postgres.Tag<T>
+declare function Postgres<T extends Postgres.CustomTypeList = {}>(url: string, options?: Postgres.Options<T>): Postgres.Tag<T>
+
+interface BaseOptions<T extends Postgres.CustomTypeList> {
+  /** Postgres ip address or domain name */
+  host?: string;
+  /** Postgres server port */
+  port?: number;
+  /** Name of database to connect to */
+  database?: string;
+  /** Username of database user */
+  username?: string;
+  /** True; or options for tls.connect */
+  ssl?: boolean | import('tls').TlsOptions;
+  /** Max number of connections */
+  max?: number;
+  /** Idle connection timeout in seconds */
+  timeout?: number;
+  /** Array of custom types; see more below */
+  types?: T;
+  /** Defaults to console.log */
+  onnotice?: (notice: string) => void;
+  /** (key; value) when server param change */
+  onparameter?: (key: string, value: any) => void;
+  /** Is called with (connection; query; parameters) */
+  debug?: (connection: number, query: string, parameters: any[]) => void;
+  /** Transform hooks */
+  transform?: {
+    /** Transforms incoming column names */
+    column?: (column: string) => any;
+    /** Transforms incoming row values */
+    value?: (value: any) => any;
+    /** Transforms entire rows */
+    row?: (row: Postgres.Row) => any;
+  };
+  /** Connection parameters */
+  connection?: Postgres.ConnectionVariables;
+}
+
+type UnwrapPromiseArray<T> = T extends any[] ? {
+  [k in keyof T]: T[k] extends Promise<infer R> ? R : T[k]
+} : T;
+
+declare namespace Postgres {
+
+  function toPascal(str: string): string;
+  function toCamel(str: string): string;
+  function toKebab(str: string): string;
+
+  interface ConnectionVariables {
+    /** Default application_name */
+    application_name?: string;
+    /** Other connection parameters */
+    [name: string]: any;
+  }
+  
+  interface Options<T extends CustomTypeList> extends BaseOptions<T> {
+    /** unix socket path (usually '/tmp') */
+    path?: string;
+    /** Password of database user */
+    pass?: string | (() => string | Promise<string>);
+    /** Password of database user */
+    password?: Options<T>['pass']; // FIXME Is it a doc error ?
```

Comment:
Looking at the library code, both are valid and `pass` is checked before `password`.
There are more options that have "aliases", e.g. `hostname` and `host`, `database` and `db`, `user` and `username`. All these are combined in ParsedOptions into `host`, `database`, `user`, `pass`. In such case, should these be reflected as well in the typings?

Reply:
I mostly infered the typings with the `tests/index.js` file, so it's possible some part was forgotten :smile: 
Thanks for the help :wink: